### PR TITLE
Handle all matching project excludes

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1558)

### CODE REVIEW

File: reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt [hunk 1]
```diff
@@ -23,6 +23,7 @@ import com.here.ort.model.Identifier
 import com.here.ort.model.OrtIssue
 import com.here.ort.model.OrtResult
 import com.here.ort.model.RuleViolation
+import com.here.ort.model.config.Excludes
```
Comment (mnonnenmacher) [inline reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt]: Commit message nit: "noll" -> "null"

File: model/src/main/kotlin/config/Excludes.kt [hunk 2]
```diff
@@ -60,18 +60,26 @@ data class Excludes(
...
      * [project] specific excludes from [projects].
      */
-    fun findScopeExcludes(scope: Scope, project: Project, ortResult: OrtResult) =
-        scopes.filter { it.matches(scope.name) } +
-                (findProjectExclude(project, ortResult)?.scopes?.filter { it.matches(scope.name) } ?: emptyList())
+    fun findScopeExcludes(scope: Scope, project: Project, ortResult: OrtResult): List<ScopeExclude> {
+        val globalScopeExcludes = scopes.filter { it.matches(scope.name) }
+        val projectScopeExcludes = findProjectExcludes(project, ortResult)
+            .map { it.scopes }
+            .flatten()
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/config/Excludes.kt]: You can shorten this to `flatMap { it.scopes }`.

File: model/src/main/kotlin/config/Excludes.kt [hunk 3]
```diff
@@ -113,14 +121,13 @@ data class Excludes(
...
-        scopes.any { it.matches(scope.name) }
-                || findProjectExclude(project, ortResult)?.scopes?.any { it.matches(scope.name) } ?: false
+        findScopeExcludes(scope, project, ortResult).isNotEmpty()
 
     /**
      * Map the project excludes by the identifiers of the provided [projects].
      */
     fun projectExcludesById(projects: Set<Project>, ortResult: OrtResult) =
-        projects.associate { Pair(it.id, findProjectExclude(it, ortResult)) }
+        projects.associate { Pair(it.id, findProjectExcludes(it, ortResult).singleOrNull()) }
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/config/Excludes.kt]: Wouldn't the correct fix be to not use `singleOrNull` here? Or do you plan to fix that separately?

File: model/src/test/kotlin/config/ExcludesTest.kt [hunk 4]
```diff
@@ -93,17 +96,18 @@ class ExcludesTest : WordSpec() {
             }
         }
 
-        "findProjectExclude" should {
-            "return null if there is no matching project exclude" {
+        "findProjectExcludes" should {
+            "return empty list there is no matching project exclude" {
```
Comment (mnonnenmacher) [inline model/src/test/kotlin/config/ExcludesTest.kt]: Missing "if" after "list".

File: reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt [hunk 5]
```diff
@@ -95,11 +95,11 @@ class ReportTableModelMapper(private val resolutionProvider: ResolutionProvider)
...
-            val projectExclude = excludes.findProjectExclude(project, ortResult)?.let { exclude ->
-                // Only add the project exclude to the model if the whole project is excluded. If only parts of the
-                // project are excluded this information will be stored in the rows of the affected dependencies.
-                exclude.takeIf { it.isWholeProjectExcluded }
-            }
+            // Only add the project exclude to the model if the whole project is excluded. If only parts of the
+            // project are excluded this information will be stored in the rows of the affected dependencies.
+            val projectExclude = excludes.findProjectExcludes(project, ortResult)
+                .filter { it.isWholeProjectExcluded }
+                .singleOrNull()
```
Comment (mnonnenmacher) [inline reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt]: We should show all matching project excludes in the report, but I'm fine with fixing that separately.

File: reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt (hunk 1 above)
Comment (fviernau) [reply reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt]: Done.
